### DISCUSSION TITLE
Theming strategy?

### DISCUSSION THREAD

**Discussion by shah** [lumeland/lume#100] [Q&A] [answered]
I was curious what the approach should be to translate a theme from Hugo to Lume. Specifically, we built out a Salesforce Lightning Design System (SLDS) based Hugo theme that we'd like to start with but I did not see any theming opinions in the Lume documentation. 

How should we go about porting our [SLDS Hugo Theme](https://github.com/netspective-studios/hugo-theme-sf-lightning) over to Lume? There's nothing special about our theme, it's basically SLDS components make accessible through Hugo and we'd want to do the same thing with Lume.

I get the sense that [Lume Layouts](https://lumeland.github.io/creating-pages/layouts/) is where we'd build our "theme" or is there another strategy that the community suggests?

**Comment by oscarotero** [comment, marked as answer]
Lume is unopinionated and doesn't have a "standard" theming system. This Hugo theme could be ported by translating the layouts files to nunjucks (or other template system available) and store them in `_includes`. The other files are just static files so it can be copied as is.

**Reply by shah** [reply]
Great, that's what I was thinking as well. Although, I would suggest that -- in case you're competing with Hugo, Zola, WordPress, etc. you should definitely consider an opinionated theming approach because that creates a user community faster. Perhaps if you just treat themes as plugins then you'd be able leave the base untouched but allow easy install of themes as plugins.

Hugo a little while started using Go Modules as themes and life has been easier ever since ... every theme is basically a Golang module and a union file system architecture is used to combine content with presentation. Decoupling of themes through modules has been great and worth considering for Lume.
